app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
```

Comment:
I DID return 'error' or 'disabled' in an earlier commit (I've squashed it now), but gillux recommended I do it this way.